src/weather.cpp

```diff
@@ -511,7 +511,7 @@ std::string weather_forecast(radio_tower tower)
     double high = -100.0;
     double low = 100.0;
     // TODO wind direction and speed
-    int last_hour = calendar::turn - (calendar::turn % HOURS(1));
+    int last_hour = calendar::turn - (calendar::is_time_for(HOURS(1)));
```

Comment:
See above. This is to make `last_hour` the time when the current hour started, e.g. if turn is "09:44:12", `last_hour` should become "09:00:00".